File: Beecrowd/1117 Validação de Nota/main.py
A = 11
B = 11
while A < 0 or A > 10:
  A = float(input())
  if A < 0 or A > 10:
    print("nota invalida")
	
while B < 0 or B > 10:
  B = float(input())
  if B < 0.0 or B > 10.0:
    print("nota invalida")

media = (A + B) / 2
print("media = %0.2f" %media)

File: Beecrowd/1117 Validação de Nota/main.cpp
#include <iostream>
 
using namespace std;
 
int main() {
 
	double A, B, media;
	
	do{
		cin >> A;
		if(A < 0.0 || A > 10.0)
			cout << "nota invalida" << endl;
	}while(A < 0 || A > 10);
	
	do{
		cin >> B;
		if(B < 0.0 || B > 10.0)
			cout << "nota invalida" << endl;
	}while(B < 0 || B > 10);

	media = (A + B) / 2;
	
	cout <<"media = " << media << endl;
 
    return 0;
}